### DISCUSSION TITLE
How do I bind commands to F1-F12?

### DISCUSSION THREAD

**Discussion by zhutao1212** [helix-editor/helix#6357] [Q&A]
I tried "C-F10" = ":run-shell-command cargo run" but it didn't work. What if I used a normal key like "C-t" = ":run-shell-command cargo run"?

**Comment by the-mikedavis** [comment]
That syntax is correct. For example, this binds Ctrl F2 to `toggle_comments` (Ctrl c by default):

```toml
# ~/.config/helix/config.toml
[keys.normal]
C-F2 = "toggle_comments"
```

Your terminal emulator or operating system might be handling the C-F10 keypress instead of passing it to Helix though.

Also note that `:run-shell-command` is meant for small shell invocations and not to be used for building or running projects. That would be covered by #1976 and should be run in a separate terminal pane until then.
